--- a/net/minecraft/world/entity/animal/Bee.java
+++ b/net/minecraft/world/entity/animal/Bee.java
@@ -469,7 +_,7 @@
          return false;
       } else {
          BlockEntity blockentity = this.f_19853_.m_7702_(this.f_27698_);
-         return blockentity != null && blockentity.m_58903_() == BlockEntityType.f_58912_;
+         return blockentity instanceof BeehiveBlockEntity;
       }
    }
 
@@ -477,7 +_,7 @@
       return this.m_27921_(8);
    }
 
-   void m_27919_(boolean p_27920_) {
+   public void m_27919_(boolean p_27920_) {
       if (p_27920_) {
          this.m_27853_();
       }
@@ -489,7 +_,7 @@
       return this.m_27921_(4);
    }
 
-   private void m_27925_(boolean p_27926_) {
+   public void m_27925_(boolean p_27926_) {
       this.m_27832_(4, p_27926_);
    }
 
