--- a/net/minecraft/world/damagesource/EntityDamageSource.java
+++ b/net/minecraft/world/damagesource/EntityDamageSource.java
@@ -37,6 +_,12 @@
       return !itemstack.m_41619_() && itemstack.m_41788_() ? new TranslatableComponent(s + ".item", p_19397_.m_5446_(), this.f_19391_.m_5446_(), itemstack.m_41611_()) : new TranslatableComponent(s, p_19397_.m_5446_(), this.f_19391_.m_5446_());
    }
 
+   // CraftBukkit start
+   public Entity getProximateDamageSource() {
+      return super.m_7639_();
+   }
+   // CraftBukkit end
+
    public boolean m_7986_() {
       return this.f_19391_ instanceof LivingEntity && !(this.f_19391_ instanceof Player);
    }
