--- a/net/minecraft/world/level/BaseSpawner.java
+++ b/net/minecraft/world/level/BaseSpawner.java
@@ -30,23 +_,23 @@
    private static final Logger f_45441_ = LogManager.getLogger();
    private static final int f_151303_ = 1;
    private static WeightedRandomList<SpawnData> f_151304_ = WeightedRandomList.m_146332_();
-   private int f_45442_ = 20;
+   public int f_45442_ = 20;
    private WeightedRandomList<SpawnData> f_45443_ = f_151304_;
    private SpawnData f_45444_ = new SpawnData();
    private double f_45445_;
    private double f_45446_;
-   private int f_45447_ = 200;
-   private int f_45448_ = 800;
-   private int f_45449_ = 4;
+   public int f_45447_ = 200;
+   public int f_45448_ = 800;
+   public int f_45449_ = 4;
    @Nullable
    private Entity f_45450_;
-   private int f_45451_ = 6;
-   private int f_45452_ = 16;
-   private int f_45453_ = 4;
+   public int f_45451_ = 6;
+   public int f_45452_ = 16;
+   public int f_45453_ = 4;
    private final Random f_151305_ = new Random();
 
    @Nullable
-   private ResourceLocation m_151332_(@Nullable Level p_151333_, BlockPos p_151334_) {
+   public ResourceLocation m_151332_(@Nullable Level p_151333_, BlockPos p_151334_) {
       String s = this.f_45444_.m_47265_().m_128461_("id");
 
       try {
@@ -127,11 +_,12 @@
                   entity.m_7678_(entity.m_20185_(), entity.m_20186_(), entity.m_20189_(), p_151312_.f_46441_.nextFloat() * 360.0F, 0.0F);
                   if (entity instanceof Mob) {
                      Mob mob = (Mob)entity;
-                     if (!mob.m_5545_(p_151312_, MobSpawnType.SPAWNER) || !mob.m_6914_(p_151312_)) {
+                     if (!net.minecraftforge.event.ForgeEventFactory.canEntitySpawnSpawner(mob, p_151312_, (float)entity.m_20185_(), (float)entity.m_20186_(), (float)entity.m_20189_(), this)) {
                         continue;
                      }
 
                      if (this.f_45444_.m_47265_().m_128440_() == 1 && this.f_45444_.m_47265_().m_128425_("id", 8)) {
+                        if (!net.minecraftforge.event.ForgeEventFactory.doSpecialSpawn(mob, p_151312_, (float)entity.m_20185_(), (float)entity.m_20186_(), (float)entity.m_20189_(), this, MobSpawnType.SPAWNER))
                         ((Mob)entity).m_6518_(p_151312_, p_151312_.m_6436_(entity.m_142538_()), MobSpawnType.SPAWNER, (SpawnGroupData)null, (CompoundTag)null);
                      }
                   }
@@ -272,4 +_,12 @@
    public double m_45474_() {
       return this.f_45446_;
    }
+
+   @Nullable
+   public Entity getSpawnerEntity() {
+      return null;
+   }
+
+   @Nullable
+   public net.minecraft.world.level.block.entity.BlockEntity getSpawnerBlockEntity(){ return null; }
 }
