--- a/net/minecraft/world/level/chunk/ChunkGenerator.java
+++ b/net/minecraft/world/level/chunk/ChunkGenerator.java
@@ -212,7 +_,38 @@
       }
    }
 
+   public void addDecorations(WorldGenRegion p_62168_, StructureFeatureManager p_62169_, boolean vanilla) {
+      if (vanilla) {
+         m_7399_(p_62168_, p_62169_);
+      }
+      org.bukkit.World world = p_62168_.getMinecraftWorld().getWorld();
+      // only call when a populator is present (prevents unnecessary entity conversion)
+      if (world.getPopulators().size() != 0) {
+         org.bukkit.craftbukkit.generator.CraftLimitedRegion limitedRegion = new org.bukkit.craftbukkit.generator.CraftLimitedRegion(p_62168_);
+         int x = p_62168_.m_143488_().f_45578_;
+         int z = p_62168_.m_143488_().f_45579_;
+         for (org.bukkit.generator.BlockPopulator populator : world.getPopulators()) {
+            WorldgenRandom seededrandom = new WorldgenRandom();
+            seededrandom.m_64690_(p_62168_.m_7328_(), x, z);
+            populator.populate(world, seededrandom, x, z, limitedRegion);
+         }
+         limitedRegion.saveEntities();
+         limitedRegion.breakLink();
+      }
+      // CraftBukkit end
+   }
+
    public abstract void m_7338_(WorldGenRegion p_62170_, ChunkAccess p_62171_);
+
+   // CraftBukkit start - spilt surface and bedrock generation code
+   public WorldgenRandom buildSurface(WorldGenRegion regionlimitedworldaccess, ChunkAccess iChunkAccess) {
+      throw new UnsupportedOperationException("Methode not overridden");
+   }
+
+   public void buildBedrock(ChunkAccess iChunkAccess, Random random) {
+      throw new UnsupportedOperationException("Methode not overridden");
+   }
+   // CraftBukkit end
 
    public void m_6929_(WorldGenRegion p_62167_) {
    }
