--- a/net/minecraft/world/level/chunk/ChunkBiomeContainer.java
+++ b/net/minecraft/world/level/chunk/ChunkBiomeContainer.java
@@ -104,4 +_,13 @@
       int k = p_62135_ & f_62106_;
       return this.f_62112_[j << f_62109_ + f_62109_ | k << f_62109_ | i];
    }
+
+   // CraftBukkit start
+   public void setBiome(int x, int y, int z, Biome biome) {
+      int i = x & f_62106_;
+      int j = Mth.m_14045_(y, 0, this.f_156120_);
+      int k = z & f_62106_;
+      this.f_62112_[j << f_62109_ + f_62109_ | k << f_62109_ | i] = biome;
+   }
+   // CraftBukkit end
 }
