--- a/net/minecraft/world/level/chunk/LevelChunkSection.java
+++ b/net/minecraft/world/level/chunk/LevelChunkSection.java
@@ -13,7 +_,7 @@
    public static final int f_156455_ = 16;
    public static final int f_156456_ = 16;
    public static final int f_156457_ = 4096;
-   private static final Palette<BlockState> f_62967_ = new GlobalPalette<>(Block.f_49791_, Blocks.f_50016_.m_49966_());
+   public static final Palette<BlockState> f_62967_ = new GlobalPalette<>(Block.f_49791_, Blocks.f_50016_.m_49966_());
    private final int f_62968_;
    private short f_62969_;
    private short f_62970_;
